Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -116,6 +118,44 @@ public AndConstraint<XDocumentAssertions> NotBeEquivalentTo(XDocument unexpected
             return new AndConstraint<XDocumentAssertions>(this);
         }
 
+        /// <summary>
+        /// Asserts that document is valid by XSD schema.
+        /// </summary>
+        /// <param name="xsd">XDS schema.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        /// <returns>Instance of <see cref="AndConstraint{T}"/>.</returns>
+        public AndConstraint<XDocumentAssertions> BeValidByXsd(string xsd, string because = "", params object[] becauseArgs)
+        {
+            var isValid = true;
+            var errorMessage = default(string);
+            var schemaSet = new XmlSchemaSet();
+
+            using var textReader = new StringReader(xsd);
+            using var xmlReader = XmlReader.Create(textReader);
+
+            schemaSet.Add(string.Empty, xmlReader);
+            schemaSet.Compile();
+
+            Subject.Validate(schemaSet, (o, e) =>
```

Comment:
Will this event only ever be raised once? If not, consider collecting all the error messages.